### PR TITLE
Add basic testing config and lint previous errors

### PR DESCRIPTION
**Pull request by inge4pres**
## Summary

Added jest config to run tests, further testing will be applied when types will be introduced.
We now have applied most of the linting needed to have pre-commit hook succeed.

Please review commit-by-commit.
One more thing we could do is have a single file for the topn searches, not sure we need 5 different files.
Thoughts?

### CODE REVIEW

**Comment by jbcrail** [comment]
I agree that separate files for each route is not necessary. I think we can further consolidate down to 4 files, e.g.:
```
load_flamechart.ts
load_topn.ts
search_flamechart.ts
search_topn.ts
```

**Review by jbcrail — APPROVED** [review]
Looks good :+1:

**Comment by inge4pres** [comment]
> I agree that separate files for each route is not necessary. I think we can further consolidate down to 4 files, e.g.:
> 
> ```
> load_flamechart.ts
> load_topn.ts
> search_flamechart.ts
> search_topn.ts
> ```

Nice, thanks 🙏🏼 will rebase and resolve conflict reducing the number of files

File: src/plugins/profiling/server/routes/load_topn.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      return response.ok({ body: transformFlamechart(src) });
+    }
+  );
+}
+
+export function registerTraceEventsTopNThreadsRoute(router: IRouter<DataRequestHandlerContext>) {
+  const paths = getLocalRoutePaths();
+  router.get(
+    {
+      path: paths.TopNThreads,
```
Comment (rockdaboot) [inline src/plugins/profiling/server/routes/load_topn.ts:113]: Does it make sense to reduce code duplication ? E.g. with a function that takes 'router', 'path' and 'fixture_prefix` ?

**Review by rockdaboot — APPROVED** [review]

File: src/plugins/profiling/server/routes/load_topn.ts (hunk 1 above)
Comment (inge4pres) [reply src/plugins/profiling/server/routes/load_topn.ts:113]: yeah I had thought about it 🤔

**Pull request merged**